stl/inc/format

```diff
@@ -665,7 +665,7 @@ public:
     }
 
     constexpr void _On_type(_CharT _Type) {
-        _Specs._Type = static_cast<_CharT>(_Type);
+        _Specs._Type = static_cast<char>(_Type);

```

Comment:
Can we cite some standardese here? It is a non obvious cast

Reply:
I don't think I have any specific standardese, except for the fact that all types are representable by `char`s and that the type of `_Type` is `char`. I am not really sure what the correct move here is. Maybe we should make `_Type` `_CharT` and then handle anything outside of the possible types when writing with specs? @barcharcraz you authored this code, what was the reasoning for the original `static_cast`?

Reply:
That smeels like ab bug. Why is it a template if `_CharT` is always `char`? What happens if we have a context with `wchar_t`

Reply:
I do think that `_Type` should be `char`, but we should check that the value we set it to is actually an ascii character.

Reply:
I am going to delay this change until we get the proper machinery in place to validate setting the type.